client/src/app/_components/groups/GroupItem.tsx

```diff
@@ -44,7 +33,16 @@ const GroupItem = ({ currentGroup, creatorId }: GroupItemProps) => {
 	// };
 
 	const handleClick = (e: Event | undefined) => {
-		void router.push(`/groups/${group?.name}`);
+		//At this point we have all group information from firebase and IPFS
+		//Pass to reduce need to query?
+		void router.push({
+			pathname: `/groups/${firebaseGroup.id}`,
+			query: {
```

Comment:
Yeah we can fetch from the db but the GroupItem object has already had the db fetched from. So in this instance we already have most of the group info available to us so I thought passing params in query would be better as reduces the number of db calls.

On top of this, the way the custom hooks work in tRPC means that we can't actually chain hooks together inside the same component (this is an issue I spoke to you about previous regarding displaying the name of the product in te dropdown for groups). We can have it so that we only have group ID in URL but that will mean we will have to re-query to get group data on page open, this will have to be in its own component, then we'll need a seperate component to then get the product info.

Up to you what direction you want to take it in dude.

Reply:
I seeeee, okay let's leave it like that for now, but if ever users would want to share the URL to the group, ideally would just be with UUID and user B that will navigate directly to that group and fetch from firebase. For now let's stay with current approach, thanks for explaining 👍

Reply:
Yeah for sure we can think about changing stuff up later down the line

Reply:
Have reduced it to groupId and hash, so looks cleaner. Now we have description too actually have to re-fetch group details anyway so stripped some shit out.
![Screenshot 2024-05-11 at 09 20 46](https://github.com/KarolDevLtd/Sliced.club/assets/36128399/51f0757c-55c6-4c73-831a-f1ffd04cfaf0)